src/index.ts

```diff
@@ -698,6 +704,104 @@ class Logging {
+   * is terminated, it will continue reading logs.
+   *
+   * @method Logging#tailEntries
+   * @param {TailEntriesRequest} [query] Query object for tailing entries.
+   * @returns {DuplexStream} A duplex stream that emits TailEntriesResponses
+   * containing an array of {@link Entry} instances.
+   *
+   * @example
+   * const {Logging} = require('@google-cloud/logging');
+   * const logging = new Logging();
+   *
+   * logging.tailEntries()
+   *   .on('error', console.error)
+   *   .on('data', resp => {
+   *     console.log(resp.entries);
+   *     console.log(resp.suppressionInfo);
+   *   })
+   *   .on('end', function() {
+   *     // All entries retrieved.
+   *   });
+   *
+   * //-
+   * // If you anticipate many results, you can end a stream early to prevent
+   * // unnecessary processing and API requests.
+   * //-
+   * logging.getEntriesStream()
+   *   .on('data', function(entry) {
+   *     this.end();
+   *   });
+   */
+  tailEntries(options: TailEntriesRequest = {}) {
+    const userStream = streamEvents<Duplex>(pumpify.obj());
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    let gaxStream: ClientDuplexStream<any, any>;
+
+    (userStream as AbortableDuplex).abort = () => {
+      if (gaxStream && gaxStream.cancel) {
+        gaxStream.cancel();
+      }
+    };
+
+    const transformStream = through.obj((data, _, next) => {
+      next(
+        null,
+        (() => {
+          const formattedEntries: Entry[] = [];
+          data.entries.forEach((entry: google.logging.v2.LogEntry) => {
+            formattedEntries.push(Entry.fromApiResponse_(entry));
+          });
+          const resp: TailEntriesResponse = {
+            entries: formattedEntries,
+            suppressionInfo: data.suppressionInfo,
+          };
+          return resp;
+        })()
+      );
+    });
+
+    this.auth.getProjectId().then(projectId => {
+      this.projectId = projectId;
+
+      if (options.log) {
+        if (options.filter) {
+          options.filter = `(${options.filter}) AND logName="${Log.formatName_(
+            this.projectId,
+            options.log
+          )}"`;
+        } else {
+          options.filter = `logName="${Log.formatName_(
+            this.projectId,
+            options.log
+          )}"`;
+        }
+      }
+      options.resourceNames = arrify(options.resourceNames);
+      options.resourceNames.push(`projects/${this.projectId}`);
+      const writeOptions = {
+        resourceNames: options.resourceNames,
+        ...(options.filter && {filter: options.filter}),
+        ...(options.bufferWindow && {bufferwindow: options.bufferWindow}),
+      };
+
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      if (!(global as any).GCLOUD_SANDBOX_ENV) {
+        gaxStream = this.loggingService.tailLogEntries(options.gaxOptions);
+        // Write can only be called once in a single tail streaming session.
+        gaxStream.write(writeOptions);
```

Comment:
that answers my question, if I'm following do you write options regarding how the stream will be tailed, and then listen on it for events?

Reply:
Correct. We have the option of writing options that define filters & bufferwindow for what/how to retrieve the logs.